server/src/main/java/io/spine/server/aggregate/AggregateCommandEndpoint.java

```diff
@@ -46,13 +47,33 @@ I handle(AggregateRepository<I, A> repository, CommandEnvelope command) {
         return endpoint.handle();
     }
 
-
     static <I, A extends Aggregate<I, ?, ?>>
     AggregateCommandEndpoint<I, A>
     of(AggregateRepository<I, A> repository, CommandEnvelope command) {
         return new AggregateCommandEndpoint<>(repository, command);
     }
 
+    @Override
+    protected void deliverNowTo(A aggregate) {
+        ensureNotDuplicate(aggregate);
```

Comment:
This isn't a responsibility of an endpoint. It's a responsibility of an `Entity`. It must not be possible to "forget" to call `ensureNotDuplicate`; it should be an implementation feature of `Aggregate`. 

With the current approach it's quite possible to feed a duplicate command.

Reply:
We‘ve discussed this in a personal meeting and decided the check must be performed at the endpoint. This is one of the points in the list of tasks higher in this conversation.
Aggregate doesn’t know anything about its repository, placing the check inside of aggregate would make it harder to check for duplicates if we need more then the events played when loading the aggregate.